src/shutdown/shutdown.c

```diff
@@ -255,6 +256,42 @@ static void sync_with_progress(void) {
         (void) kill(pid, SIGKILL);
 }
 
+static int read_current_sysctl_printk_log_level(void) {
+        _cleanup_free_ char *sysctl_printk_vals = NULL, *sysctl_printk_curr = NULL;
+        unsigned current_lvl = 0;
+        const char *p;
+        int r;
+
+        r = sysctl_read("kernel/printk", &sysctl_printk_vals);
+        if (r < 0)
+                return log_debug_errno(r, "Cannot read sysctl kernel.printk: %m");
+
+        p = sysctl_printk_vals;
+        r = extract_first_word(&p, &sysctl_printk_curr, NULL, 0);
+        if (r > 0)
+                r = safe_atou(sysctl_printk_curr, &current_lvl);
+        else if (r == 0)
+                r = -EINVAL;
+
+        if (r < 0)
+                return log_debug_errno(r, "Unexpected sysctl kernel.printk content: %s", sysctl_printk_vals);
+
+        return current_lvl;
+}
+
+static void update_sysctl_printk_log_level(int min_level) {
+        /* Set the logging level to be able to see messages with log level smaller or equal to min_level */
+
+        int current_lvl = read_current_sysctl_printk_log_level();
+        if (current_lvl >= 0 && current_lvl <= min_level) {
+                char buf[DECIMAL_STR_MAX(int)];
+                xsprintf(buf, "%d\n", min_level + 1);
+                int r = sysctl_write("kernel/printk", buf);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to bump kernel.printk to %d: %m", min_level + 1);
```

Comment:
not that it matters much, but I think instead of outputing "min_level + 1" here, I think it would make more sense to output "buf" directly so that the increment is only done once

Reply:
I am fine with this suggestion, the only problem is that `buf` contains a new line...
Maybe I can create a temporary variable with the result of the addition?